### PR TITLE
Enable AzureOp packaging

### PR DESCRIPTION
**Pull request by RandySheriffH**
- Enable Azure Ops to be built in packaging pipelines by a control flag
- Implement Azure invokers for text endpoints

### CODE REVIEW

File: install_deps.bat
Comment (wenbingl) [inline install_deps.bat:1]: move them into tool folder, the root folder is a little crowded now.

File: install_deps.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (wenbingl) [inline install_deps.sh:1]: move this file into tool folder?

File: install_deps.bat [hunk 2]
```diff
@@ -0,0 +1,7 @@
+@echo off
+if "%OCOS_ENABLE_AZURE%"=="1" (
+  pip install cmake
+  if not exist "C:\Program Files\Miniconda3\python3.exe" (
```
Comment (wenbingl) [inline install_deps.bat:4]: %ProgramFiles% ?

File: includes/custom_op_lite.h [hunk 3]
```diff
@@ -62,6 +65,7 @@ class TensorBase {
   bool is_input_;
   std::optional<std::vector<int64_t>> shape_;
   ONNXTensorElementDataType type_ = ONNX_TENSOR_ELEMENT_DATA_TYPE_UNDEFINED;
+  const char* mem_type_ = "Cpu";
```
Comment (wenbingl) [inline includes/custom_op_lite.h:68]: why not std::string?

**Review by wenbingl — COMMENTED** [review]
Overall, it looks good to me.
Did you try to run the pipeline with BuildAzureOp=1

**Comment by RandySheriffH** [comment]
> Overall, it looks good to me. Did you try to run the pipeline with BuildAzureOp=1

Yes, it builds fine with minor things to follow up with.

File: install_deps.bat
Comment (RandySheriffH) [reply install_deps.bat:1]: done.

File: install_deps.sh (hunk 1 above)
Comment (RandySheriffH) [reply install_deps.sh]: done.

File: install_deps.bat (hunk 2 above)
Comment (RandySheriffH) [reply install_deps.bat]: done.

**Review by wenbingl — APPROVED** [review]

**Pull request merged**